### ISSUE TITLE
Update to 2.0.6 breaks mitmproxy

### ISSUE THREAD

**Issue by mkrsfcmp** [cuckoosandbox/cuckoo#2332]
##### My issue is:

Upgrade to 2.0.6 requires specific python version modules which are not compatible with old mitmproxy. And new mitmproxy is not compatible with python 2.x.
It can probably be worked around by installing mitmproxy with sufficiently new python3 alongside system-wide python2 but it's not easy in case of - for example - CentOS7.

##### My Cuckoo version and operating system are:
CentOS7, cuckoo 2.0.6

##### This can be reproduced by:
Updating cuckoo with pip.
During the update you get:
"mitmproxy 0.18.3 has requirement cryptography<1.6,>=1.3, but you'll have cryptography 2.2.2 which is incompatible.
mitmproxy 0.18.3 has requirement Flask<0.12,>=0.10.1, but you'll have flask 0.12.2 which is incompatible.
mitmproxy 0.18.3 has requirement jsbeautifier<1.7,>=1.6.3, but you'll have jsbeautifier 1.6.2 which is incompatible.
mitmproxy 0.18.3 has requirement requests<2.12,>=2.9.1, but you'll have requests 2.13.0 which is incompatible."
Since cuckoo is python2-only and current mitmproxy is python3-only, it gets tricky to run mitmproxy with python3 and cuckoo with python2.

**Comment by RicoVZ** [comment]
Hi mkrsfcmp,

Thanks for posting an issue.

The best course of action is installing Cuckoo in a virtualenv. This should prevent interference with other Python applications that have specific requirements.

```
$ mkdir somedir
$ cd somedir
$ virtualenv .
$ source bin/activate
$ pip install -U cuckoo
```

**Comment by mkrsfcmp** [comment]
Hi Rico
I understand and I do have a virtualenv installation. But it's way more complicated to have to keep part of the installation within the virtualenv (the cuckoo part) and another part (the mitmproxy) outside, or even in another virtualenv. Up to 2.0.5 cuckoo worked with legacy mitmproxy so there was no hassle. Now it won't. And keeping two very different versions of python can be troublesome.
It's not that I expect a quick fix and complete cuckoo rewrite to Python 3.6 ;-). I just posted it as a sign that there is an issue in case someone else stumbles upon it and that it can be worked around albeit not easily (especially in CentOS, where installing Python 3.6 for mitmproxy requires enabling SCL and doing some other strange exercises).
BTW, it seems that upgrade also broke something with immediate cuckoo dependencies. I'm getting "Failed to run the processing module "Static"" for jobs.
Shall I create a separate issue for that or shall we close this one and I'll open another?

**Comment by jbremer** [comment]
FWIW I'm awaiting the `mitmproxy` team to see if they're interested to merge the following as part of a `0.18.2.1` release, which would resolve this exact issue; https://github.com/mitmproxy/mitmproxy/compare/master...jbremer:0.18.2-cuckoo
Meanwhile you can install that version of mitmproxy in your virtualenv. You're correct that Python 3 is still a long stretch for now :-)
